python/ray/tune/examples/bohb_example.py

```diff
@@ -50,6 +50,13 @@ def load_checkpoint(self, checkpoint_dir):
 
 
 if __name__ == "__main__":
+    import sys
+
+    import pytest
+
+    if sys.version_info >= (3, 12):
```

Comment:
What version of Python does our CI use? IIRC we used the minimum supported version, but I might be wrong.

Reason I ask is if we don't use 3.12, I'm worried this will get missed in the future, and it might be better to let it explicitly fail and we can make a decision on what to do with the integration.

Reply:
@can-anyscale Is CI using py39?

Reply:
> Is CI using py39?

CI uses py39 yes

> How often does CI run with py312?

You decide; currently we only run tests with py39 versions; historically for other python versions we do a one-fix effort during upgrade but do not run tests continuously

> Is it ok if these tests explicitly fail for 312?

We can choose to intentionally NOT run/fix it for python 3.12 now and let it fail explicitly when CI turns py39 to py312

Reply:
> We can choose to intentionally NOT run/fix it for python 3.12 now and let it fail explicitly when CI turns py39 to py312

Explicitly failing on py312 makes sense to me!

Reply:
sweet, you can just note in https://docs.google.com/spreadsheets/d/1VIpZdrp24CiWkI13EE__REIFlVn0lDiSBLuCN5cyOP8/edit?gid=2025695384#gid=2025695384 that we intentionally ignore the tests for python 3.12 and mark it as green, thankks